src-tests/test-rpc.lisp

```diff
@@ -142,23 +142,59 @@
 (defun error-with-serious-condition ()
   (error 'serious-condition))
 
-(deftest test-serious-condition ()
+(defun error-with-error-condition ()
+  (error "owie!"))
+
+(deftest test-error-handling ()
+  "Test that various types of processing errors are handled correctly."
+  (dolist (rpc-method '("error-with-serious-condition"
+                        "error-with-error-condition"
+                        "unknown-method"))
+    (with-unique-rpc-address (addr)
+      (let* ((server-function
+               (lambda ()
+                 (let ((dt (rpcq:make-dispatch-table)))
+                   (rpcq:dispatch-table-add-handler dt 'error-with-serious-condition)
+                   (rpcq:start-server :dispatch-table dt
+                                      :listen-addresses (list addr)))))
+             (server-thread (bt:make-thread server-function)))
+        (sleep 1)
+        (unwind-protect
+             ;; hook up the client
+             (rpcq:with-rpc-client (client addr)
+               ;; send a communique
+               (signals rpc-error
+                 (rpcq:rpc-call client rpc-method)))
+          ;; kill the server thread
+          #+ccl
+          (loop :while (bt:thread-alive-p server-thread)
+                :do (sleep 1) (bt:destroy-thread server-thread))
+          #-ccl
+          (bt:destroy-thread server-thread))))))
+
+(deftest test-invalid-rpc-request ()
+  "Test that invalid RPC requests are handled correctly."
   (with-unique-rpc-address (addr)
     (let* ((server-function
              (lambda ()
                (let ((dt (rpcq:make-dispatch-table)))
-                 (rpcq:dispatch-table-add-handler dt 'error-with-serious-condition)
-                 (rpcq:start-server :timeout 5
-                                    :dispatch-table dt
+                 (rpcq:dispatch-table-add-handler dt 'test-method)
+                 (rpcq:start-server :dispatch-table dt
                                     :listen-addresses (list addr)))))
            (server-thread (bt:make-thread server-function)))
       (sleep 1)
       (unwind-protect
-           ;; hook up the client
-           (rpcq:with-rpc-client (client addr)
-             ;; send a communique
-             (signals rpc-error
-               (rpcq:rpc-call client "error-with-serious-condition")))
+           ;; The invalid request will be ignored, resulting in no response being sent, so we
+           ;; instead test that the request times out.
```

Comment:
I could additionally check the logs here to make sure the expected error is logged if folks feel like the timeout check is insufficient.

Reply:
this is done now